generator/templates/server.go.tmpl

```diff
@@ -278,6 +430,30 @@ func recoverError{{ $serviceName }}(v interface{}, rawStack []byte) *{{ $service
 	}
 }
 
+func (s *{{ $serviceName }}) evalCEL(expr string, vars []cel.EnvOption, args map[string]any, outType reflect.Type) (any, error) {
+	env, err := s.env.Extend(vars...)
+	if err != nil {
+		return nil, err
+	}
+	expr = strings.Replace(expr, "$", "__ARG__", -1)
```

Comment:
nit, might be better if we check the expr contains __ARG__ before replacing $ with it, even though that is quite unlikely users use it...

Reply:
Thank you. I will verify this in the resolver phase. I also assume that we always evaluate valid expressions in the generator phase ( because generator phase must be evaluated after resolver phase ).